src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java

```diff
@@ -0,0 +1,291 @@
+        assertEquals(coursePaginationList, data.getCoursePaginationList());
+        assertEquals(feedbackSessions, data.getFeedbackSessions());
+        expectedNextPageLink = "javascript:;";
+        expectedPreviousPageLink = data.getInstructorCommentsLink() + "&courseid=" + course1.id;
+        assertEquals(data.getNextPageLink(), expectedNextPageLink);
+        assertEquals(data.getPreviousPageLink(), expectedPreviousPageLink);        
+        assertEquals(numberOfPendingComments, data.getNumberOfPendingComments());
+        assertFalse(data.isDisplayArchive());
+        assertFalse(data.isViewingDraft());
+        
+        /******************** Assertions for data structures ********************/
+        
+        actualCommentsForStudentsTables = data.getCommentsForStudentsTables();
+        assertEquals(1, actualCommentsForStudentsTables.size());
+        assertEquals(0, actualCommentsForStudentsTables.get(0).getRows().size());
+    }
+    
+    private List<CommentAttributes> getCommentsForGiverInCourse(String giverEmail, String courseId) {
+        List<CommentAttributes> commentsForGiverInCourseList = new ArrayList<CommentAttributes>();
+        for (CommentAttributes comment : dataBundle.comments.values()) {
+            if (comment.giverEmail.equals(giverEmail) && comment.courseId.equals(courseId)) {
+                commentsForGiverInCourseList.add(comment);
+            }
+        }
+        return commentsForGiverInCourseList;
+    }
+
+    private List<FeedbackSessionAttributes> getFeedbackSessionsForCourse(String courseId) {
+        List<FeedbackSessionAttributes> feedbackSessionsInCourse = new ArrayList<FeedbackSessionAttributes>();
+        for (FeedbackSessionAttributes feedbackSession : dataBundle.feedbackSessions.values()) {
+            if (feedbackSession.courseId.equals(courseId)) {
+                feedbackSessionsInCourse.add(feedbackSession);
+            }
+        }
+        return feedbackSessionsInCourse;
+    }
+
+    private List<InstructorAttributes> getInstructorsInCourse(String courseId) {
+        List<InstructorAttributes> instructorsInCourse = new ArrayList<InstructorAttributes>();
+        for (InstructorAttributes instructor : dataBundle.instructors.values()) {
+            if (instructor.courseId.equals(courseId)) {
+                instructorsInCourse.add(instructor);
+            }
+        }
+        return instructorsInCourse;
+    }
+
+    private List<StudentAttributes> getStudentsInCourse(String courseId) {
+        List<StudentAttributes> studentsInCourse = new ArrayList<StudentAttributes>();
+        for (StudentAttributes student : dataBundle.students.values()) {
+            if (student.course.equals(courseId)) {
+                studentsInCourse.add(student);
+            }
+        }
+        return studentsInCourse;
+    }
+    
+    private void isCommentsForStudentsTablesEqual(CommentsForStudentsTable expected, CommentsForStudentsTable actual) {
```

Comment:
Maybe `checkCommentsForStudentsTablesEqual` can be a better name? `is.....` is only for returning `boolean` value.